src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,135 @@
+package duke;
+
+public class Parser {
+    public static Command parseCommand(String userInput) throws DukeException{
+        String firstWordOfInput = userInput.split(" ")[0];
+        CommandType commandType;
+        String[] additionalParameters = null;
+        switch (firstWordOfInput) {
+        case CommandInputs.ADD_TODO_COMMAND_INPUT:
+            commandType = CommandType.ADD_TODO_COMMAND;
+            additionalParameters = getToDoAdditionalParameters(userInput);
+            break;
+        case CommandInputs.ADD_DEADLINE_COMMAND_INPUT:
+            commandType = CommandType.ADD_DEADLINE_COMMAND;
+            additionalParameters = getDeadlineAdditionalParameters(userInput);
+            break;
+        case CommandInputs.ADD_EVENT_COMMAND_INPUT:
+            commandType = CommandType.ADD_EVENT_COMMAND;
+            additionalParameters = getEventAdditionalParameters(userInput);
+            break;
+        case CommandInputs.LIST_TASKS_COMMAND_INPUT:
+            commandType = CommandType.LIST_TASKS_COMMAND;
+            break;
+        case CommandInputs.MARK_COMMAND_INPUT:
+            commandType = CommandType.MARK_TASK_COMMAND;
+            additionalParameters = getTaskNumbers(userInput);
+            break;
+        case CommandInputs.UNMARK_COMMAND_INPUT:
+            commandType = CommandType.UNMARK_TASK_COMMAND;
+            additionalParameters = getTaskNumbers(userInput);
+            break;
+        case CommandInputs.DELETE_COMMAND_INPUT:
+            commandType = CommandType.DELETE_TASK_COMMAND;
+            additionalParameters = getTaskNumbers(userInput);
+            break;
+        case CommandInputs.END_PROGRAM_COMMAND_INPUT:
+            commandType = CommandType.END_PROGRAM_COMMAND;
+            break;
+        default:
+            throw new InvalidCommandException();
+        }
+        return new Command(commandType,additionalParameters);
+    }
+
+    protected static String[] getTaskNumbers(String userInput) throws DukeException{
+        userInput = userInput.trim();
+        int indexOfFirstSpace = userInput.indexOf(" ");
+        if(indexOfFirstSpace == -1 || indexOfFirstSpace == userInput.length() - 1){
+            throw new TaskNumberException();
+        }
+        String[] taskNumbers = userInput.substring(indexOfFirstSpace).trim().split(" ");
+        if(taskNumbers.length != 1){
+            throw new TaskNumberException();
+        }
+        return taskNumbers;
```

Comment:
Consider including comments on what conditions are being checked in these statements, as it may not be very clear to developers who are unfamiliar with this program